docs/proposals/dynamic-openshift-settings-configmaps.md

```diff
@@ -0,0 +1,55 @@
+# Changing OpenShift settings dynamically through ConfigMaps
+
+## Problem
+
+Currently, there are settings contained withing the master-config.yaml file that
+are of value to an unpriviledged user. Specifically, the `autoProvisionEnabled`
+boolean under `jenkinsPipelineConfig` is only available to users with the
+ability to write to the master-config.
+
+## Design proposal:
+
+- In each project there can exist a ConfigMap annotated with
+  'build.openshift.io/configuration'
```

Comment:
can there exist more than one configmap with this annotation?  i'm thinking yes, but that makes it the responsibility of code consuming these values to find *all* the relevant configmaps and resolve the possibility that the same key could be represented multiple times in different configmaps (and then how do you decide which value to use?).

i'm ok with saying "if you do that, we don't guarantee behavior" but it should be called out in the proposal.